api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/ListOfPatientsCurrentlyOnArtWithoutTbScreeningCohortQueries.java

```diff
@@ -0,0 +1,580 @@
+  }
+
+  /**
+   * <b>Technical Specs</b>
+   *
+   * <blockquote>
+   *
+   * <p><b>Total number of patients on the List (TB_NSCRN_FR2) </b>
+   *
+   * <p>Number of Patients Currently on ART without TB screening and at least one Clinical
+   * Consultation in last 6 months
+   *
+   * </blockquote>
+   *
+   * @return {@link CohortDefinition}
+   */
+  public CohortDefinition
+      getPatientsCurrentlyOnArtWithoutTbScreeningAndWithoutClinicalConsultationInLast6Months() {
+
+    CompositionCohortDefinition composition = new CompositionCohortDefinition();
+    addParameters(composition);
+    composition.setName("Currently on ART without TB Screening and Without Clinical Consultation");
+
+    CohortDefinition currentlyOnArtWithoutTbScreening =
+        getPatientsCurrentlyOnArtWithoutTbScreening();
+    CohortDefinition withClinicalConsultationInLast6Months =
+        getPatientsWithClinicalConsultationInLast6Months();
+
+    composition.addSearch(
+        "onArtWithoutScreening",
+        EptsReportUtils.map(
+            currentlyOnArtWithoutTbScreening, "endDate=${endDate},location=${location}"));
+    composition.addSearch(
+        "withConsultation",
+        EptsReportUtils.map(
+            withClinicalConsultationInLast6Months,
+            "startDate=${endDate-6m},endDate=${endDate},location=${location}"));
+
+    composition.setCompositionString("onArtWithoutScreening AND NOT withConsultation");
+
+    return composition;
+  }
+
+  public DataDefinition getArtStartDate() {
+
+    SqlPatientDataDefinition sqlPatientDataDefinition = new SqlPatientDataDefinition();
+    sqlPatientDataDefinition.setName("Get ART Start Date");
+    sqlPatientDataDefinition.addParameter(new Parameter("location", "Location", Location.class));
+    sqlPatientDataDefinition.addParameter(new Parameter("endDate", "End Date", Location.class));
+
+    sqlPatientDataDefinition.setQuery(commonQueries.getARTStartDate(true));
+
+    return sqlPatientDataDefinition;
+  }
+
+
+  /**
+   * <b>Technical Specs</b>
+   *
+   * <blockquote>
+   * <p>Type of Dispensation registered on the most recent Clinical Consultation (value of Column Q) registered on Ficha Clínica – MasterCard by Report End Date
+   * <p>Possible values are:
+   * <ul>
+   *     <li>DM</li>
+   *     <li>DT</li>
+   *     <li>DS</li>
+   *
+   * </ul>   *
+   * <p>Note: For Patients without Type of Dispensation informed, the corresponding cell in the excel file will show NA</p>
+   *
+   * </blockquote>
+   *
+   * @return {@link DataDefinition}
+   */
+  public DataDefinition getDispensationTypeOnClinicalAndPediatricEncounter() {
+
+    SqlPatientDataDefinition sqlPatientDataDefinition = new SqlPatientDataDefinition();
+    sqlPatientDataDefinition.setName("Dispensation Type on Encounter 6");
+    sqlPatientDataDefinition.addParameter(new Parameter("location", "Location", Location.class));
+    sqlPatientDataDefinition.addParameter(new Parameter("endDate", "End Date", Location.class));
```

Comment:
Date taking Location object here is NOT accurate

Reply:
Fixed